easyapplybot.py

```diff
@@ -251,31 +282,45 @@ def get_easy_apply_button(self):
 		return EasyApplyButton
 
 
+	def is_jsonable(self, x):
+		try:
+			json.dumps(x)
+			return True
+		except:
+			return False
+
 	def send_resume(self):
 		def is_present(button_locator):
-			return len(self.browser.find_elements(button_locator[0],
-													 button_locator[1])) > 0
+			return (len(self.browser.find_elements(button_locator[0], button_locator[1])) > 0)
 
 		try:
-			time.sleep(random.uniform(1.5, 2.5))
-			next_locater = (By.CSS_SELECTOR,
-							"button[aria-label='Continue to next step']")
-			review_locater = (By.CSS_SELECTOR,
-							"button[aria-label='Review your application']")
-			submit_locater = (By.CSS_SELECTOR,
-								"button[aria-label='Submit application']")
-			submit_application_locator = (By.CSS_SELECTOR,
-									"button[aria-label='Submit application']")
-			error_locator = (By.CSS_SELECTOR,
-							"p[data-test-form-element-error-message='true']")
-			upload_locator = (By.CSS_SELECTOR, "input[name='file']")
+
+			time.sleep(3)
+			log.info("Attempting to send resume")
+			#TODO These locators are not future proof. These labels could easily change. Ideally we would search for contained text;
+			# was unable to get it to work using XPATH and searching for contained text
+			upload_locator = (By.CSS_SELECTOR, "label[aria-label='DOC, DOCX, PDF formats only (2 MB).']")
```

Comment:
```suggestion
			
```

Reply:
See comment about creating a state machine.

Im having a hard time keeping up with what exactly is happening, needs to happen, and Im not sure if you and I are seeing the same screens/logic.

Reply:
I agree, the codebase needs a major refactoring. 

There are several things that don't make sense anymore. Just to make a clear example, all of this is happening inside a function named `send_resume`